tests/matrix_free/matrix_vector_hp.cc

```diff
@@ -45,50 +45,11 @@ class MatrixFreeTestHP
               const Vector<Number> &                       src,
               const std::pair<unsigned int, unsigned int> &cell_range) const
   {
-    // ask MatrixFree for cell_range for different orders
-    std::pair<unsigned int, unsigned int> subrange_deg =
-      data.create_cell_subrange_hp(cell_range, 1);
-    if (subrange_deg.second > subrange_deg.first)
-      helmholtz_operator<dim, 1, Vector<Number>, 2>(data,
-                                                    dst,
-                                                    src,
-                                                    subrange_deg);
-    subrange_deg = data.create_cell_subrange_hp(cell_range, 2);
-    if (subrange_deg.second > subrange_deg.first)
-      helmholtz_operator<dim, 2, Vector<Number>, 3>(data,
-                                                    dst,
-                                                    src,
-                                                    subrange_deg);
-    subrange_deg = data.create_cell_subrange_hp(cell_range, 3);
-    if (subrange_deg.second > subrange_deg.first)
-      helmholtz_operator<dim, 3, Vector<Number>, 4>(data,
-                                                    dst,
-                                                    src,
-                                                    subrange_deg);
-    subrange_deg = data.create_cell_subrange_hp(cell_range, 4);
-    if (subrange_deg.second > subrange_deg.first)
-      helmholtz_operator<dim, 4, Vector<Number>, 5>(data,
-                                                    dst,
-                                                    src,
-                                                    subrange_deg);
-    subrange_deg = data.create_cell_subrange_hp(cell_range, 5);
-    if (subrange_deg.second > subrange_deg.first)
-      helmholtz_operator<dim, 5, Vector<Number>, 6>(data,
-                                                    dst,
-                                                    src,
-                                                    subrange_deg);
-    subrange_deg = data.create_cell_subrange_hp(cell_range, 6);
-    if (subrange_deg.second > subrange_deg.first)
-      helmholtz_operator<dim, 6, Vector<Number>, 7>(data,
-                                                    dst,
-                                                    src,
-                                                    subrange_deg);
-    subrange_deg = data.create_cell_subrange_hp(cell_range, 7);
-    if (subrange_deg.second > subrange_deg.first)
-      helmholtz_operator<dim, 7, Vector<Number>, 8>(data,
-                                                    dst,
-                                                    src,
-                                                    subrange_deg);
+    if (data.get_cell_active_fe_index(cell_range) > 0) // exclude FE_Nothing
+      helmholtz_operator<dim, -1, Vector<Number>, 0>(data,
```

Comment:
I would prefer keep the test as it is to check if the template parameters are still picked up correctly in the hp case. We already have a non-templated test: https://github.com/dealii/dealii/blob/3bba74760bff6724e1766bf22e4ee06eaf98bfc1/tests/matrix_free/matrix_vector_hp_no_template.cc#L56C9-L56C39, which is, as far as I remember correctly, a copy of the current test.

Reply:
But we should then simplify the test to only run in 2D and only use 2 or at most 3 polynomial degrees that get their code compiled. If you agree, I can make that change.